scripts/src/ecosystem_scripts/validate_registry.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python
+"""Valididate packages' meta.yaml and generate an output directory with json/images to be uploaded on github pages"""
+
+from __future__ import annotations
+
+import argparse
+import json
+import os
+import shutil
+import sys
+from importlib.resources import files
+from logging import basicConfig, getLogger
+from pathlib import Path
+from textwrap import dedent
+from typing import TYPE_CHECKING
+
+import httpx
+import jsonschema
+import yaml
+from PIL import Image
+from rich.logging import RichHandler
+from rich.traceback import install
+
+if TYPE_CHECKING:
+    from collections.abc import Generator, Iterable, Mapping
+else:
+    from collections.abc import Iterable
+
+log = getLogger(__name__)
+
+# Constants
+HTTP_OK = 200
+HTTP_NOT_FOUND = 404
+IMAGE_SIZE = 512
+
+
+class LinkChecker:
+    """Track known links and validate URLs."""
+
+    def __init__(self):
+        self.known_links: set[str] = set()
+
+    def check_and_register(self, url: str, context: str) -> None:
+        """Check if URL is duplicate, validate it exists, and register it.
+
+        Parameters
+        ----------
+        url
+            The URL to check and register
+        context
+            Context information for error messages (e.g., file being validated)
+        """
+        if url in self.known_links:
+            msg = f"{context}: Duplicate link: {url}"
+            raise ValueError(msg)
+
+        response = httpx.head(url, follow_redirects=True)
+        if response.status_code != HTTP_OK:
+            msg = f"URL {url} is not reachable (error {response.status_code}). "
```

Comment:
Do we want to error early or just collect the failures and raise an error at the end? It's a bit annoying if you have to keep rerunning it to collect all failures.

Reply:
imlemented